### PR TITLE
update validate designs to provide --supported feature

### PR DESCRIPTION
**Pull request by vishnuchebrolu**
We have made changes in existing commandline flow. As now host is expecting path should be coming with -p otherwise design will fail.

1.	Change in the commandline flow : 
a.	Older flow:
Verify test : ./validate.exe <path to platform test folder>
b.	New flow :
Verify test : ./validate.exe -p <path to platform test folder>   
2.	New –supported feature usage
a.	Verify test : ./validate.exe -s -p <path to platform test folder>

### CODE REVIEW

**Comment by stsoe** [comment]
@vishnuchebrolu Shouldn't all these validate tests be able to build on Windows?

File: tests/validate/bandwidth_test/src/host.cpp
Comment (AShivangi) [inline tests/validate/bandwidth_test/src/host.cpp:46]: initialize

File: tests/validate/bandwidth_test/src/host.cpp
Comment (AShivangi) [inline tests/validate/bandwidth_test/src/host.cpp:67]: Instead of creating a flag, you can simply check if the str is empty or not
```suggestion
    if (test_path.empty()) {
```

**Review by AShivangi — CHANGES_REQUESTED** [review]
Hi @vishnuchebrolu, 
Thanks for making the changes. I'll approve the PR once you address the minor comments. 

Can you also add the usage (old and new) to the description of this PR? You can simply copy over what you sent in the email

**Comment by AShivangi** [comment]
Marking this PR "do not merge" until we have the accompanying xbutil fix available as this change will break the existing xbutil validate

File: tests/validate/bandwidth_test/src/host.cpp
Comment (vishnuchebrolu) [reply tests/validate/bandwidth_test/src/host.cpp]: updated.

File: tests/validate/bandwidth_test/src/host.cpp
Comment (vishnuchebrolu) [reply tests/validate/bandwidth_test/src/host.cpp]: updated.

**Comment by vishnuchebrolu** [comment]
> @vishnuchebrolu Shouldn't all these validate tests be able to build on Windows?

@stsoe I guess these validate tests will be able to build on Windows, but I have not tested on Windows OS.

**Comment by stsoe** [comment]
> > @vishnuchebrolu Shouldn't all these validate tests be able to build on Windows?
> 
> @stsoe I guess these validate tests will be able to build on Windows, but I have not tested on Windows OS.

They will not, they are including and using header files not available on windows.  I noticed because you switched to getopt.h, and also include other OS specific header files.

**Review by AShivangi — APPROVED** [review]
Looks good!

**Comment by vishnuchebrolu** [comment]
> > > @vishnuchebrolu Shouldn't all these validate tests be able to build on Windows?
> > 
> > 
> > @stsoe I guess these validate tests will be able to build on Windows, but I have not tested on Windows OS.
> 
> They will not, they are including and using header files not available on windows. I noticed because you switched to getopt.h, and also include other OS specific header files.

@stsoe I have updated the designs to not use getopt.h to support both linux and windows OS. Please review the PR.

**Comment by stsoe** [comment]
@vishnuchebrolu Don't let windows hold you up if these tests are not applicable to windows just yet. 
There are includes in the tests that are not available on windows, so I am sure the code will not compile with Visual Studio.

**Comment by AShivangi** [comment]
Hi @stsoe, 
I requested Vishnu to make the change for two reasons:
1. Proactively trying to keep the code OS-agnostic to avoid a major clean up when we need these tests on windows
2. We are anyways waiting for my accompanying xbutil validate change :)

**Comment by AShivangi** [comment]
xbutil change: https://github.com/Xilinx/XRT/pull/5648

**Comment by dayeh-xilinx** [comment]
retest this please.

**Pull request merged**
